packages/manager/src/features/CloudPulse/Utils/CloudPulseWidgetUtils.ts

```diff
@@ -194,33 +195,52 @@ export const generateGraphData = (props: graphDataOptionsProps) => {
           serviceType,
           unit,
         };
-
-        const dimension = {
-          backgroundColor: color,
-          borderColor: color,
-          data: seriesDataFormatter(transformedData.values, start, end),
-          fill: widgetChartType === 'area',
-          label: getLabelName(labelOptions),
-        };
+        const labelName = getLabelName(labelOptions);
+        const data = seriesDataFormatter(transformedData.values, start, end);
+        const color = colors[index]?.Primary ?? Alias.Chart.Neutral;
+        areas.push({
+          color,
+          dataKey: labelName,
+        });
+
+        data.forEach((d) => {
+          const timestamp = d[0];
+          const value = d[1];
+          if (value !== null) {
+            dimension[timestamp] = {
+              ...dimension[timestamp],
+              [labelName]: value,
+            };
+          }
+        });
         // construct a legend row with the dimension
-        const legendRow = {
-          data: getMetrics(dimension.data as number[][]),
+        const legendRow: MetricsDisplayRow = {
+          data: getMetrics(data as number[][]),
           format: (value: number) => formatToolTip(value, unit),
           legendColor: color,
-          legendTitle: dimension.label,
+          legendTitle: labelName,
         };
         legendRowsData.push(legendRow);
-        dimensions.push(dimension);
-        today ||= isToday(start, end);
       }
     );
   }
 
+  const maxUnit = generateMaxUnit(legendRowsData, unit);
+  const dimensions: DataSet[] = Object.entries(dimension).map(
+    ([key, value]) => {
+      const rolledUpData: { [resource: string]: number } = {};
+      Object.entries(value).forEach(
```

Comment:
In this forEach loop, rolledUpData is being mutated directly. This will work, but may be using functional approach like "reduce" could make it clearer. 
`const rolledUpData = Object.entries(value).reduce(`

Reply:
thanks for suggestions